vtex-sites/nextjs.store
# chore: use transition api to defer navbar behavior

## What's the purpose of this pull request?

This PR makes updates to the `useScrollDirection` hook less frequent, which improves navbar performance

## How does it work?

It uses the new `useTransition` hook to defer state updates.

## How to test it?

Browse the deploy preview and make sure the navbar behavior on scroll stays the same as in vtexfaststore.com

## Checklist

<em>You may erase this after checking them all :wink:</em>

**PR Title and Commit Messages**
  - Available prefixes: `feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, and `test`

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Features`, `Bug fixes`, `Chore`, `Documentation`, `Style changes`, `Refactoring`, `Performance`, and `Test`
- [x] Added the component, hook, or path name in-between backticks (\`\`) - *if applicable, e.g., `ComponentName` component, `useWindowDimensions` hook*

**Dependencies**

**Documentation**
